pkg/streams/multi.go

```diff
@@ -0,0 +1,48 @@
+package streams
+
+import (
+	"io"
+	"sync"
+)
+
+// multiWriter allows caller to broadcast data to several writers.
+type multiWriter struct {
+	sync.Mutex
+	writers []io.WriteCloser
+}
+
+// Add registers one writer into MultiWriter.
+func (mw *multiWriter) Add(writer io.WriteCloser) {
+	mw.Lock()
+	mw.writers = append(mw.writers, writer)
+	mw.Unlock()
+}
+
+// Write writes data into several writers and never returns error.
+func (mw *multiWriter) Write(p []byte) (int, error) {
+	mw.Lock()
+	var evictIdx []int
+	for n, w := range mw.writers {
+		if _, err := w.Write(p); err != nil {
+			w.Close()
+			evictIdx = append(evictIdx, n)
+		}
+	}
+
+	for n, i := range evictIdx {
+		mw.writers = append(mw.writers[:i-n], mw.writers[i-n+1:]...)
+	}
+	mw.Unlock()
+	return len(p), nil
```

Comment:
if all of the writer writing bytes fails here, still return `len(p)`. Is it reasonable?

Reply:
since we don't return error here, we need to return the len(p) here. The writer hub uses io.Pipe as `io.WriteCloser`. All data are handled in memory, not the real file or network IO. If it run into a error, there is some wrong with pouch and we should restart pouchd.

I think I can add the warning message here if it run into error during writing. How do you think?